packages/backend/src/server/api/call.ts

```diff
@@ -116,7 +116,7 @@ export default async (endpoint: string, user: CacheableLocalUser | null | undefi
 
 	// API invoking
 	const before = performance.now();
-	return await ep.exec(data, user, token, ctx?.file).catch((e: Error) => {
+	return await ep.exec(data, user, token, ctx?.file, ctx?.ip, ctx?.headers).catch((e: Error) => {
```

Comment:
Maybe even already check here if IP logging is enabled. And if logging is disabled only pass `null`?

Reply:
It is necessary here because it may be used for other purposes than logging.